features/networking/multus.feature

```diff
@@ -606,3 +606,252 @@ Feature: Multus-CNI related scenarios
       | namespace isolation |
       | violat              |
     """
+ 
+  # @author anusaxen@redhat.com
+  # @case_id OCP-24490
+  @admin
+  Scenario: Pods can communicate each other with same vlan tag
+    # Make sure that the multus is enabled
+    Given the multus is enabled on the cluster
+    And I store all worker nodes to the :nodes clipboard
+    Given I have a project
+    When I run the :create admin command with:
+      | f | https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/multus-cni/NetworkAttachmentDefinitions/bridge-host-local-vlan.yaml |
+      | n | <%= project.name %>                                                                                                                               |
+    Then the step should succeed
+    #Labeling a worker node to make sure couple of future pods to be scheduled on this node only
+    When I run the :label admin command with:
+      | resource | node                    |
+      | name     | <%= cb.nodes[0].name %> |
+      | key_val  | test=worker1            |
+    Then the step should succeed
+    
+    Given I register clean-up steps:
+    """
+    When I run the :label admin command with:
+      | resource | node                    |
+      | name     | <%= cb.nodes[0].name %> |
+      | key_val  | test-                   |
+    Then the step should succeed
+    """ 
+    #Labeing another worker node to make sure 3rd future pod to be scheduled on this node only
+    When I run the :label admin command with:
+      | resource | node                    |
+      | name     | <%= cb.nodes[1].name %> |
+      | key_val  | test=worker2            |
+    Then the step should succeed
+    
+    Given I register clean-up steps:
+    """
+    When I run the :label admin command with:
+      | resource | node                    |
+      | name     | <%= cb.nodes[1].name %> |
+      | key_val  | test-            	   |
+    Then the step should succeed
+    """ 
+    #Creating first pod in vlan 100
+    When I run oc create over "https://raw.githubusercontent.com/openshift-qe/v3-testfiles/master/networking/multus-cni/Pods/generic_multus_pod_nodeselector.yaml" replacing paths:
+      | ["metadata"]["name"] | pod1-vlan100 |
+      | ["metadata"]["annotations"]["k8s.v1.cni.cncf.io/networks"]| bridgevlan100 |
+      | ["spec"]["nodeSelector"]["test"] | worker1 |
+    Then the step should succeed
+    And the pod named "pod1-vlan100" becomes ready
+    And evaluation of `pod.name` is stored in the :pod1 clipboard
+    And I execute on the pod:
+      | ifconfig | net1 |
+    Then the step should succeed
+    And evaluation of `@result[:response].match(/\d{1,3}\.\d{1,3}.\d{1,3}.\d{1,3}/)[0]` is stored in the :pod1_net1_ip clipboard
```

Comment:
btw will this work for IPv6-only clusters that we are trying to support?

Reply:
@akostadinov We are not sure about that yet. These are interfaces inside pods (multus) so its too early to say whether they will support ipv6 or not. AFAIK they might use dual stack approach..ipv4 routing via ipv6. We can see for optimization in future is its required.